### PR TITLE
ENG-8966 handle constructor race condition in channel distributer

### PR DESCRIPTION
**Pull request by vtkstef**

### CODE REVIEW

File: tests/frontend/org/voltdb/regressionsuites/TestLimitOffsetSuite.java [hunk 1]
```diff
@@ -343,10 +343,7 @@ public void testSubqueryLimit() throws Exception {
         cr = client.callProcedure("@AdHoc", "INSERT INTO R1 VALUES (15, 'ebWfhdmIZfYhRC', 23926, 3.61291695704730075889e-01); ");
         assertEquals(ClientResponse.SUCCESS, cr.getStatus());
 
-        String selectStmt = "select NUM from R1 where NUM in (select NUM from R1 where NUM <> 12 order by NUM limit 4);";
-        cr = client.callProcedure("@AdHoc", selectStmt);
-        assertEquals(ClientResponse.SUCCESS, cr.getStatus());
-
+        String selectStmt = "select NUM from R1 where NUM in (select NUM from R1 where NUM <> 12 order by NUM limit 4) ORDER BY NUM;";
```
Comment (akhanzode) [inline tests/frontend/org/voltdb/regressionsuites/TestLimitOffsetSuite.java]: why this change needed in this?

**Comment by akhanzode** [comment]
looks good.

**Comment by vtkstef** [comment]
--merge squashed into master and release-5.6.x
